### PR TITLE
Finish breaking apart the catalogue API terraform

### PR DESCRIPTION
**Pull request by alexwlchan**
Closes https://github.com/wellcomecollection/platform/issues/4386, and paves the way to update it all to terraform 0.12, one stack at a time.

All applied, no changes, as expected.

I’ve moved the shared infra into a module "shared", which exports the pieces the staging/prod APIs need. There's then a new module "catalogue_api_stack" which absorbs all the duplicated parts.

### CODE REVIEW

File: api/terraform/catalogue_api_prod/locals.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+locals {
+  namespace        = "catalogue_api"
+  namespace_hyphen = "${replace(local.namespace,"_","-")}"
+
+  vpc_id                         = "${data.terraform_remote_state.shared_infra.catalogue_vpc_id}"
+  private_subnets                = "${data.terraform_remote_state.shared_infra.catalogue_vpc_private_subnets}"
+  gateway_server_error_alarm_arn = "${data.terraform_remote_state.shared_infra.gateway_server_error_alarm_arn}"
+
+  # This is the Zone ID for wellcomecollection.org in the routemaster account.
```
Comment (kenoir) [inline api/terraform/catalogue_api_prod/locals.tf]: I think you might be able to do that.

Comment (kenoir) [reply api/terraform/catalogue_api_prod/locals.tf]: _now_

Comment (alexwlchan) [reply api/terraform/catalogue_api_prod/locals.tf]: :D

**Comment by alexwlchan** [comment]
The formatting task has passed; merging the rest rather than waiting for the Scala builds to run.

**Pull request merged**